### ISSUE TITLE
Bug: `maxRecommendedTextures` ignores the current context (DOM vs worker)

### ISSUE THREAD

**Issue by gabrielecirulli** [pixijs/pixijs#10546]
This issue appears for the first time in `8.1.4`. Introduced by https://github.com/pixijs/pixijs/pull/10538

### Current Behavior

In [rendering/batcher/shared/const.ts](https://github.com/pixijs/pixijs/blob/e3d1df40b3b6fafd2e1d20db341fe0fd0c424b38/src/rendering/batcher/shared/const.ts) there's a call to `maxRecommendedTextures()`. Since it's exported as a const it runs every time the code is loaded. https://github.com/pixijs/pixijs/blob/e3d1df40b3b6fafd2e1d20db341fe0fd0c424b38/src/rendering/batcher/shared/const.ts#L3

`maxRecommendedTextures()` calls `getTestContext()`:
https://github.com/pixijs/pixijs/blob/e3d1df40b3b6fafd2e1d20db341fe0fd0c424b38/src/rendering/renderers/shared/texture/utils/maxRecommendedTextures.ts#L10-L15

`getTestContext()` uses a `DOMAdapter` to create a `canvas`. 
https://github.com/pixijs/pixijs/blob/e3d1df40b3b6fafd2e1d20db341fe0fd0c424b38/src/rendering/renderers/gl/shader/program/getTestContext.ts#L13-L23

This crashes the code inside a worker:
```
Uncaught ReferenceError: document is not defined
    at Object.createCanvas (BrowserAdapter.mjs:4:20)
    at N0 (getTestContext.mjs:7:37)
    at Cp (maxRecommendedTextures.mjs:5:14)
    at const.mjs:4:22
```

### Expected Behavior

The code should not crash inside a worker.

### Steps to Reproduce

Use sample: https://github.com/gabrielecirulli/pixi-max-recommended-textures-worker-crash

### Environment

- **`pixi.js` version**: `8.1.4`
- **Browser & Version**: Arc `Version 1.43.0 (50018)` `Chromium Engine Version 125.0.6422.60`

### Possible Solution

If possible, check whether the current environment has DOM or is a worker before creating the canvas.

**Comment by gabrielecirulli** [comment]
There are other uses of `getTestContext()` that may cause the same issue within a worker. Worth looking at all of them?

https://github.com/search?q=repo%3Apixijs%2Fpixijs%20getTestContext()&type=code

**Comment by GoodBoyDigital** [comment]
sorry about that! Fixed by #10558
